vulture_os/toolkit/api_parser/varonis/varonis.py

```diff
@@ -0,0 +1,308 @@
+                    },
+                    {
+                        "path":"Alert.Status.ID",
+                        "operator":"In",
+                        "values":[
+                            {"Alert.Status.ID": "1", "displayValue": "New"},
+                            {"Alert.Status.ID": "2", "displayValue": "Under Investigation"},
+                            {"Alert.Status.ID": "3", "displayValue": "Closed"},
+                            {"Alert.Status.ID": "4", "displayValue": "Escalated"},
+                            {"Alert.Status.ID": "5", "displayValue": "Auto-Resolved"}
+                        ]
+                    },
+                    {
+                        "path":"Alert.AggregationFilter",
+                        "operator":"Equals",
+                        "values":[ { "Alert.AggregationFilter": 1 } ]
+                    }
+                ]
+            }
+        }
+
+        (e_terminate, content) = self.varonis_api_search(rows, query)
+
+        alerts = []
+        for r in content.get("rows"):
+            alert = {content["columns"][i]: r[i] for i in range(len(r))}
+            alerts.append(alert)
+
+        # Destroying the search
+        self.varonis_api_search_terminate(e_terminate)
+
+        return alerts
+
+    def get_alerted_events(self, alert_id):
+        rows = {
+            "columns": self.ALL_COLUMNS_EVENT,
+            "filter":[],
+            "grouping":"",
+            "ordering":[]
+        }
+
+        query = {
+            "entityName": "Event",
+            "filter": {
+                "filterOperator": 0,
+                "filters": [
+                    {
+                        "path": "Event.TimeUTC",
+                        "operator": "LastDays",
+                        "values": [ { "Event.TimeUTC": 365, "displayValue": 365 } ]
+                    },
+                    {
+                        "path": "Event.Alert.ID",
+                        "operator": "IncludesAny",
+                        "values": [ { "Event.Alert.ID": alert_id, "displayValue": alert_id } ]
+                    },
+                ]
+            },
+            "timeZone": "Europe/Paris"
+        }
+
+        (e_terminate, content) = self.varonis_api_search(rows, query)
+        events = []
+        for r in content["rows"]:
+            event = {content["columns"][i]: r[i] for i in range(len(r))}
+            events.append(event)
+
+        # Destroying the search
+        self.varonis_api_search_terminate(e_terminate)
+
+        return events
+
+    def retrieve_events_from_alerts(self, alerts_ids: list[str]):
+        events = []
+        for alert_id in alerts_ids:
+            events.extend(self.get_alerted_events(alert_id))
+        return events
+
+    def get_logs(self, since, to):
+        alerts = self.get_alerts(since, to)
+        if alerts:
+            self.last_api_call = datetime.strptime(alerts[-1]["Alert.TimeUTC"], "%Y-%m-%dT%H:%M:%S")
```

Comment:
Bad practice to change this kind of attribute in a function named `get_*()`, especially for the `last_api_call` parameter
also, be careful as `datetime.strptime` returns a timezone-naive timestamp here!